CRM/Custom/Form/Group.php

```diff
@@ -286,77 +235,41 @@ public function buildQuickForm() {
         'name' => ts('Cancel'),
       ],
     ];
-    if (!$this->_isGroupEmpty && !empty($this->_subtypes)) {
+    if (!$this->_isGroupEmpty && !empty($this->_values['extends_entity_column_value'])) {
       $buttons[0]['class'] = 'crm-warnDataLoss';
     }
     $this->addButtons($buttons);
   }
 
   /**
-   * Set default values for the form. Note that in edit/view mode
-   * the default values are retrieved from the database
-   *
-   *
+   * Set default values for the form.
    * @return array
-   *   array of default values
    */
-  public function setDefaultValues() {
-    $defaults = &$this->_defaults;
-    $this->assign('showMaxMultiple', TRUE);
+  public function setDefaultValues(): array {
+    $defaults = &$this->_values;
     if ($this->_action == CRM_Core_Action::ADD) {
       $defaults['weight'] = CRM_Utils_Weight::getDefaultWeight('CRM_Core_DAO_CustomGroup');
 
-      $defaults['is_multiple'] = $defaults['min_multiple'] = 0;
       $defaults['is_active'] = $defaults['is_public'] = $defaults['collapse_adv_display'] = 1;
       $defaults['style'] = 'Inline';
     }
-    elseif (empty($defaults['max_multiple']) && !$this->_isGroupEmpty) {
-      $this->assign('showMaxMultiple', FALSE);
-    }
-
-    if (($this->_action & CRM_Core_Action::UPDATE) && !empty($defaults['is_multiple'])) {
-      $defaults['collapse_display'] = 0;
-    }
-
-    if (isset($defaults['extends'])) {
-      $extends = $defaults['extends'];
-      unset($defaults['extends']);
-
-      $defaults['extends'][0] = $extends;
-
-      if (!empty($this->_subtypes)) {
-        $defaults['extends'][1] = $this->_subtypes;
-      }
-      else {
-        $defaults['extends'][1] = [0 => ''];
-      }
-
-      if ($extends == 'Relationship' && !empty($this->_subtypes)) {
-        $relationshipDefaults = [];
-        foreach ($defaults['extends'][1] as $donCare => $rel_type_id) {
-          $relationshipDefaults[] = $rel_type_id;
-        }
-
-        $defaults['extends'][1] = $relationshipDefaults;
-      }
```

Comment:
This is why the PR manages to *add* functionality and metadata while not overall increasing SLOC. I deleted a bunch of code like this which:
1. Does nothing.
2. Has done nothing for over a decade.
3. Looks scary so everyone was afraid to touch it.